corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -401,6 +400,53 @@ hqDefine("cloudcare/js/form_entry/entrycontrols_full", function () {
     MultiSelectEntry.prototype = Object.create(EntryArrayAnswer.prototype);
     MultiSelectEntry.prototype.constructor = EntryArrayAnswer;
 
+    /**
+     * Represents a select2 for multiple options
+     */
+    function MultiDropdownEntry(question, options) {
+        var self = this;
+        MultiSelectEntry.call(this, question, options);
+        self.templateType = 'multidropdown';
+        self.placeholderText = gettext('Please choose an item');
+
+        self.helpText = function () {
+            return "";
+        };
+
+        self.options = ko.computed(function () {
+            return _.map(question.choices(), function (choice, idx) {
+                return {
+                    text: choice,
+                    id: idx + 1,
+                };
+            });
+        });
+
+        self.renderSelect2 = function () {
```

Comment:
To DRY this up with the select2 code in `DropdownEntry`, you could make a module-level `select2ify` function that accepts the entry and any extra select2 options as parameters and then adds `renderSelect2` and `afterRender` to the entry. `main.js` does something like this with [eventize](https://github.com/dimagi/commcare-hq/blob/228c3b9f20a9779a8f843b68685ce2b1a8a7d7d2/corehq/apps/hqwebapp/static/hqwebapp/js/main.js#L20).